### PR TITLE
Fix issue: bulk counter feature cannot compile on platforms having no sai_bulk_object_get_stats/sai_bulk_object_clear_stats

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
Why I did this?

bulk counter feature uses two new SAI API sai_bulk_object_get_stats/sai_bulk_object_clear_stats. Some vendors have no such API implemented in their SAI code. Need a way to make compilation pass.

How I fix this?

1. Add AC_CHECK_FUNCS for sai_bulk_object_get_stats/sai_bulk_object_clear_stats which generates macro HAVE_SAI_BULK_OBJECT_CLEAR_STATS/HAVE_SAI_BULK_OBJECT_GET_STATS in config.h.
2. Use ifdef to avoid compilation failure.

### CODE REVIEW

**Review by jimmyzhai — DISMISSED** [review]

**Comment by Junchao-Mellanox** [comment]
Hi @kcudnik , can we ignore the coverage checker? I don't think we can cover that.

**Comment by kcudnik** [comment]
no we can't merge without coverage fail, since it will show up on monthly report and i will get into trouble based on that, this test must pass

**Comment by kcudnik** [comment]
why you need to add ifdef for those stats ?

**Comment by kcudnik** [comment]
> Some vendors do not have `sai_bulk_object_get_stats` and `sai_bulk_object_clear_stats`, it would cause compile issue. So, I have to use ifdef to make those vendor pass the compilation. Could you suggest a way to cover the #ifdef?

but why this is vendor specific? you are adding this in virutal switch not for specific vendor are you?
if you want to test specific fidef, then extra tests would need to be added for example in unitests/vslib/ and new executable would need to be added where you specify extra defines to c compiler

**Review by kcudnik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@Junchao-Mellanox could you please check the LGTM failure?

**Comment by Junchao-Mellanox** [comment]
> @Junchao-Mellanox could you please check the LGTM failure?

Not an issue introduced by this PR, I have sent email to sonic admin add cc'ed you.

**Comment by kcudnik** [comment]
its probably missing library for swss-common on lgtm container build docker needs to be updated

**Comment by kcudnik** [comment]
i merged my fix, so if you want to retrigger you would need to submit some dummy code change since re-run from is not working somehow

**Comment by Junchao-Mellanox** [comment]
Thanks, will re-trigger

**Review by kcudnik — APPROVED** [review]

**Pull request merged**
